actions/channel_queries.ts

```diff
@@ -0,0 +1,263 @@
+function makeChannelsAndMembersQueryString(
+    operationName: ChannelQueriessOperationNames,
+    teamId: Team['id'] = '',
+) {
+    return `
+    query ${operationName} {
+        channels(userId: "me", teamId: "${teamId}", first: ${CHANNELS_MAX_PER_PAGE}) {
+          id
+          create_at: createAt
+          update_at: updateAt
+          delete_at: deleteAt
+          team {
+            id
+          }
+          type
+          display_name: displayName
+          name
+          header
+          purpose
+          last_post_at: lastPostAt
+          last_root_post_at: lastRootPostAt
+          total_msg_count: totalMsgCount
+          total_msg_count_root: totalMsgCountRoot
+          creator_id: creatorId
+          scheme_id: schemeId
+          group_constrained: groupConstrained
+          shared
+          props
+          policy_id: policyId
+          cursor
+        }
+        channelMembers(userId: "me", teamId: "${teamId}", first: ${CHANNEL_MEMBERS_MAX_PER_PAGE}) {
+          cursor
+          channel {
+            id
+          }
+          roles {
+            id
+            name
+            permissions
+          }
+          last_viewed_at: lastViewedAt
+          msg_count: msgCount
+          msg_count_root: msgCountRoot
+          mention_count: mentionCount
+          mention_count_root: mentionCountRoot
+          notify_props: notifyProps
+          last_update_at: lastUpdateAt
+          scheme_admin: schemeAdmin
+          scheme_user: schemeUser
+        }
+      }
+`;
+}
+
+export function getAllChannelsAndMembersQueryString() {
+    return JSON.stringify({
+        query: makeChannelsAndMembersQueryString(ChannelQueriessOperationNames.allChannelAndMembers),
+        operationName: ChannelQueriessOperationNames.allChannelAndMembers,
+    });
+}
+
+export function getTeamsChannelsAndMembersQueryString(teamId: Team['id']) {
+    return JSON.stringify({
+        query: makeChannelsAndMembersQueryString(ChannelQueriessOperationNames.teamsChannelAndMembers, teamId),
+        operationName: ChannelQueriessOperationNames.teamsChannelAndMembers,
+    });
+}
+
+function makeChannelsNextQueryString(
+    operationName: ChannelQueriessOperationNames,
+    cursor: string,
+    teamId: Team['id'] = '',
+) {
+    return `query ${operationName} {
+    channels(userId: "me", teamId: "${teamId}", first: ${CHANNELS_MAX_PER_PAGE}, after: "${cursor}") {
+        id
+        create_at: createAt
+        update_at: updateAt
+        delete_at: deleteAt
+        team {
+          id
+        }
+        type
+        display_name: displayName
+        name
+        header
+        purpose
+        last_post_at: lastPostAt
+        last_root_post_at: lastRootPostAt
+        total_msg_count: totalMsgCount
+        total_msg_count_root: totalMsgCountRoot
+        creator_id: creatorId
+        scheme_id: schemeId
+        group_constrained: groupConstrained
+        shared
+        props
+        policy_id: policyId
+        cursor
```

Comment:
Please use fragments to avoid repetition of the same fields: https://graphql.org/learn/queries/#fragments

Reply:
separated into two fragments. please check